Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/data/FirebaseDatabase.kt

```diff
@@ -0,0 +1,62 @@
+package com.mentalmachines.droidcon_boston.data
+
+import com.mentalmachines.droidcon_boston.data.ScheduleDatabase.ScheduleDetail
+import com.mentalmachines.droidcon_boston.data.ScheduleDatabase.ScheduleRow
+import com.mentalmachines.droidcon_boston.utils.DateUtils
+import java.text.SimpleDateFormat
+import java.util.*
+import kotlin.collections.HashMap
+
+
+open class FirebaseDatabase {
+
+    class ScheduleEvent() {
+        val speakerNames: HashMap<String, Boolean>? = null
+        val roomNames: HashMap<String, Boolean>? = null
+        val speakerIds: HashMap<String, Boolean>? = null
+        val roomIds: HashMap<String, Boolean>? = null
+        val description: String? = null;
+        val name: String? = null
+        val photo: String? = null
+        val startTime: String? = null
+
+        fun toScheduleRow(): ScheduleRow {
+            val row = ScheduleRow()
+            row.room = roomNames!!.keys.first()
+            val d = DateUtils.fromISO8601UTC(startTime)
+            val dateFormat = SimpleDateFormat("MM/dd/yyyy", Locale.US)
+            val timeFormat = SimpleDateFormat("h:mm a", Locale.US)
+
+            row.date = dateFormat.format(d)
+            row.time = timeFormat.format(d)
+            row.speakerName = speakerNames!!.keys.first()
+            row.talkDescription = description
+            row.talkTitle = name
+            // TODO: speaker photo currently not present in the events firebase child (may want to denormalize).
```

Comment:
Need to discuss this point - right now would need to do a cross-lookup just for this field in the `speakers` firebase child to obtain this data.

Reply:
Ask an ye shall receive :-)
I added a new field to the Event called "speakerNameToPhotoUrl"
It maps the speaker name to a URL w/ their photo image if it's available.
Handled this in the preprocessing script and loaded it into the Firebase DB.

Reply:
neat